src/coreclr/jit/block.h

```diff
@@ -583,12 +590,46 @@ struct BasicBlock : private LIR::Range
 
     bool IsFirstColdBlock(Compiler* compiler) const;
 
-    /* The following union describes the jump target(s) of this block */
-    union {
-        unsigned    bbJumpOffs; // PC offset (temporary only)
-        BasicBlock* bbJumpDest; // basic block
-        BBswtDesc*  bbJumpSwt;  // switch descriptor
-    };
+    unsigned GetJumpOffs() const
+    {
+        return bbJumpOffs;
+    }
+
+    void SetJumpOffs(unsigned jumpOffs)
+    {
+        bbJumpOffs = jumpOffs;
+    }
+
+    BasicBlock* GetJumpDest() const
+    {
+        return bbJumpDest;
```

Comment:
As a next step, this could have an assert for those kinds where bbJumpDest is legal:
```
assert(KindIs(BBJ_CALLFINALLY,BBJ_ALWAYS,BBJ_EHCATCHRET,BBJ_LEAVE,BBJ_COND));
```

Reply:
I wanted to include that too, but this proved to be overly aggressive in a few spots. For example, in [Compiler::fgFindBlockILOffset()](https://github.com/dotnet/runtime/blob/e146bed73a425ef275a3a57c49264f0cfc41a950/src/coreclr/jit/fgbasic.cpp#L4583), we check curr's bbJumpDest as long as it's not a switch statement. I could check if curr's jump kind is valid first to satisfy the assert in GetJumpDest(), but I'm not sure what our stance is on reducing TP to satisfy this constraint.